tinygrad/tensor.py

```diff
@@ -273,7 +273,7 @@ def data(self) -> memoryview:
     """
     assert self.dtype.fmt is not None, f"no fmt dtype for {self.dtype}"
     assert all_int(self.shape), f"no data if shape is symbolic, {self.shape=}"
-    return self._data().cast(self.dtype.fmt, self.shape)
+    return self._data().cast(self.dtype.fmt) if self.shape == (0,) else self._data().cast(self.dtype.fmt, self.shape)
```

Comment:
i think it's fine to say size 0 always return a shape (0,) memoryview, because memoryview cannot have other shape that has 0 in it. And this if should combined with `if 0 in self.shape` in `_data()`

Reply:
I have reverted the previous commit. Not sure what exactly you mean by "this if should combined with `if 0 in self.shape` in `_data()`".